packages/trader/src/Stores/Modules/Trading/Helpers/__tests__/limit-orders.spec.ts

```diff
@@ -0,0 +1,162 @@
+            take_profit: 100,
+            stop_loss: 50,
+        });
+    });
+    it('should return empty object when values remain unchanged', () => {
+        const contract_update = {
+            has_contract_update_stop_loss: false,
+            has_contract_update_take_profit: false,
+            contract_update_stop_loss: '',
+            contract_update_take_profit: '',
+            contract_info: {
+                limit_order: {
+                    take_profit: {
+                        order_amount: 100,
+                    },
+                    stop_loss: {
+                        order_amount: 50,
+                    },
+                },
+            },
+        };
+        const result = getLimitOrder(contract_update);
+        expect(result).toEqual({ stop_loss: null, take_profit: null });
+    });
+});
+
+describe('setLimitOrderBarriers', () => {
+    let contract_info: Parameters<typeof setLimitOrderBarriers>[0]['contract_info'],
+        barriers: Parameters<typeof setLimitOrderBarriers>[0]['barriers'];
+    let contract_type = 'MULTUP';
+    const mock_barrier = {
+        color: 'blue',
+        lineStyle: 'solid',
+        high: '',
+        low: '',
+        onChange: jest.fn(),
+        relative: false,
+        draggable: false,
+        hidePriceLines: false,
+        onChartBarrierChange: jest.fn(),
+        updateBarriers: jest.fn(),
+        updateBarrierShade: jest.fn(),
+        onBarrierChange: jest.fn(),
+        barrier_count: 1,
+        default_shade: 'acbd',
+    };
+
+    beforeEach(() => {
+        contract_info = {
+            growth_rate: 0.1,
+            spot_time: 1,
+            spot: 1,
+            cancellation: {},
+            error_code: '',
+            error_field: '',
+            has_error: false,
+            has_error_details: false,
+            commission: 1,
+            has_increased: false,
+            id: '',
+            message: '',
+            obj_contract_basis: {
+                text: '',
+                value: '',
+            },
+            payout: 1,
+            profit: '',
+            returns: '',
+            stake: '',
+            limit_order: {
+                take_profit: {},
+                stop_loss: {},
+                stop_out: {},
+            },
+            barrier: '2650',
+        };
+    });
+    it('should remove barriers when limit_order is empty but is_over is true and limit order barriers are supported', () => {
+        barriers = [];
+        setLimitOrderBarriers({
+            barriers,
+            contract_type,
+            contract_info,
+            is_over: true,
+        });
+        expect(barriers).toHaveLength(0);
+    });
+    it('should update barriers with changed obj_limit_order values', () => {
+        barriers = [{ ...mock_barrier, key: LIMIT_ORDER_TYPES.STOP_OUT, high: 10 }];
+        if (contract_info && contract_info.limit_order && contract_info.limit_order.stop_out !== undefined) {
```

Comment:
`contract_info?.limit_order?.stop_out `? Will it be better with optional chaining?

Reply:
done